### ISSUE TITLE
Orphaned record in comissions table

### ISSUE THREAD

**Issue by CasperBraske** [wcvendors/wcvendors#379]
Actually, I'm not sure that's the issue since I don't access to the database.

This happened after I purged all the orders from the admin panel. Then, it remained with one product in the left-hand column ("Produto"). When I deleted that product, I got what's in the image bellow.

![screenshot-2017-11-9](https://user-images.githubusercontent.com/700448/32623246-a42dfca2-c56c-11e7-827e-486b563473f9.png)

The error message is:
```
Fatal error: Uncaught Exception: Invalid order. in /home/mysite/www1/wp-content/plugins/woocommerce/includes/data-stores/abstract-wc-order-data-store-cpt.php:89 
Stack trace: 
#0 /home/mysite/www1/wp-content/plugins/woocommerce/includes/class-wc-data-store.php(147): Abstract_WC_Order_Data_Store_CPT->read(Object(WC_Order)) 
#1 /home/mysite/www1/wp-content/plugins/woocommerce/includes/abstracts/abstract-wc-order.php(109): WC_Data_Store->read(Object(WC_Order)) 
#2 /home/mysite/www1/wp-content/plugins/wc-vendors/classes/admin/class-admin-page.php(373): WC_Abstract_Order->__construct('5431') 
#3 /home/mysite/www1/wp-admin/includes/class-wp-list-table.php(1305): WCV_Admin_Page->column_default(Object(stdClass), 'order_id') 
#4 /home/mysite/www1/wp-admin/includes/class-wp-list-table.php(1242): WP_List_Table->single_row_columns(Object(stdClass)) 
#5 /home/mysite/www1/wp-admin/includes/class-wp-list-table.php(1229): WP_List_Table->single_row(Object(stdClass)) 
#6 /home/mysite/www1/wp-admin/includes/class- in /home/mysite/www1/wp-content/plugins/woocommerce/includes/data-stores/abstract-wc-order-data-store-cpt.php on line 89
```

Best regards.

**Comment by CasperBraske** [comment]
Not gone after resetting from the Status > Tools > Reset WC Vendors.

**Comment by yannbug** [comment]
I confirm that we get the same kind of error when deleting an order and then going to the commissions page. We just encountered this problem on 2 different sites.

**Comment by CasperBraske** [comment]
Reproduced locally. I'm working to fix it and perhaps send a PR.

**Comment by digitalchild** [comment]
This is because you shouldn't be deleting orders without also purging the commissions table. There is code that is supposed to fire when orders are permanently deleted to reverse the commission but not to delete it. 

Check commissions_table_sync in class-commission.php. This could be modified to allow an option to delete the relevant commission also. The reason why we don't automatically delete commissions is because this is an accounting recording and shouldn't be deleted. 

I do understand during testing phase that you will want to create and purge things, in which case you should just delete from pv_commissions; to clean it up.

**Comment by stodorovic** [comment]
It's probably fixed by https://github.com/wcvendors/wcvendors/commit/362ac4c938fa0da63deb26d81f4b38efcbf7324c

Anyway, it's recommended to use [wc_get_order](https://docs.woocommerce.com/wc-apidocs/function-wc_get_order.html) instead of `new WC_Order`. I see more similar occurrences and I'm working on PR which will fix all places.

**Comment by CasperBraske** [comment]
_There is code that is supposed to fire when orders are permanently deleted to reverse the commission but not to delete it._

Nah, yeah. This is sort of what I meant to say. "Delete the orders and mark the commission as reversed." The problem is that **all commissions** are marked reversed but that "product" is still pending there. I didn't review the code yet to see why.

**Comment by digitalchild** [comment]
All commissions? As in even entry? I haven't heard of any reports of that.

**Comment by CasperBraske** [comment]
It's fixed by the commit 362ac4c.
I will start to test my add-ons against both master here and the main release to avoid this type of issue. :)

Best regards.
